src/briefcase/integrations/docker.py

```diff
@@ -378,9 +381,377 @@ def check_output(self, args: list[SubprocessArgT], image_tag: str) -> str:
+
+        # If a TCP server is already running for the X display on the host, then start
+        # a proxy to that TCP server for the display. While most distros do not support
+        # TCP for an X display by default, X11 forwarding over SSH is facilitated by
+        # opening an SSH channel on the port for the X display; so, proxying to that
+        # channel allows the container to use the X display exposed by X11 forwarding.
+        if self._x11_is_display_tcp(display_num):
+            proxy_process = self.tools.subprocess.Popen(
+                [
+                    "socat",
+                    f"TCP-LISTEN:{proxy_display_tcp_port},reuseaddr,fork,bind=0.0.0.0",
+                    f"TCP:localhost:{6000 + display_num}",
+                ]
+            )
+
+        # Otherwise, just proxy to the UNIX socket for the display
+        elif self._x11_is_display_unix(display_num):
+            proxy_process = self.tools.subprocess.Popen(
+                [
+                    "socat",
+                    f"TCP-LISTEN:{proxy_display_tcp_port},reuseaddr,fork,bind=0.0.0.0",
+                    f"UNIX-CONNECT:/tmp/.X11-unix/X{display_num}",
+                ]
+            )
+
+        else:
+            raise BriefcaseCommandError(f"Cannot find X11 display for {DISPLAY!r}")
+
+        return proxy_process, proxy_display_num
+
+    def _x11_proxy_display_xauth_file_path(
+        self, display_num: int
+    ) -> PosixPath:  # pragma: no-cover-if-is-windows
+        """Path to the xauth database for the proxy display.
+
+        The project's build directory is used to accommodate Docker Desktop. By default,
+        Docker Desktop only allows bind mounts from directories/files within the user's
+        home directory. So, while it may be more appropriate to use a dedicated temp
+        directory, such as /tmp, this avoids having to instruct users to add that
+        directory as a source of possible bind mounts.
+        """
+        return PosixPath.cwd() / f"build/.briefcase.docker.xauth.{display_num}"
+
+    def _x11_allocate_display(self) -> int:
+        """Finds available X display number for use; raises otherwise.
+
+        A universal mechanism to request and reserve an X display does not exist. On
+        most systems, though, there will not be many active X displays. So, this
+        approach is similar to that of OpenSSH's strategy for reserving an X display
+        for X11 forwarding: simply iterate over candidate display numbers using the
+        first one that is not already is use.
+
+        This check for whether a display number is already in use is limited to
+        verifying a TCP server or UNIX socket does not already exist for the display.
+        While other transport protocols are possible, any modern system should be using
+        at least one of these transports in tandem with anything else.
+        """
+        for num in range(20, 167 + 1, 3):
```

Comment:
Are these magic numbers literally the same as those used by OpenSSH? Is there any rhyme or reason to this selection?

Reply:
These numbers are mostly arbitrary....with a goal of minimizing the time to identify an available display number.

As for OpenSSH, they run a similar [loop](https://github.com/openssh/libopenssh/blob/05dfdd5f54d9a1bae5544141a7ee65baa3313ecd/ssh/channels.c#L3559). By default (via `sshd_config`), the loop starts at display number 10 and incrementing by 1 each time, uses the first port that is available on all network interfaces. Additionally, by default, they will actually iterate this process up to nearly 1000 times to find a display 👀 

I'm definitely open to thoughts on my implementation; I've been mostly focused on getting things operational with the intention to fine tune after.

Reply:
Ok - I guess I'm just looking for a comment that explicitly says these numbers are arbitrary (with maybe an explanation of the broad strategy), rather than some deep magick in how X11 uses ports.

Reply:
Gotcha; after further consideration, I removed the particularly arbitrary step-size of 3, increased the range from 50 candidates to 250, and re-positioned the range to start at 50.

My initial consideration with a larger step size was to hopefully get out of a range collision with another process allocating displays sequentially. However, if there was a collision, the more likely case may be multiple instances of Briefcase running where a larger step-size isn't going to help anyway....unless it varied by user.

I should also mention this all has an inherent race condition where two instances of Briefcase (or something else altogether) could allocate the same display. Given the expected use-cases, this seems especially unlikely to actually happen, though.